coral-schema/src/main/java/com/linkedin/coral/schema/avro/RelToAvroSchemaConverter.java

```diff
@@ -450,31 +450,63 @@ public RexNode visitRangeRef(RexRangeRef rexRangeRef) {
 
     @Override
     public RexNode visitFieldAccess(RexFieldAccess rexFieldAccess) {
```

Comment:
just a suggestion for code readability: this method is now long with several looped and nested if-else, which makes it a bit hard to read/follow the logic (thanks for the comments!). If, on the other hand we extract each if/else body and each loop body into a separate method with expressive method name, it will be much easier to read.

Reply:
Thank you for suggestion, I extracted the second `while` loop. It is kind of hard to extract the first `while` loop because it needs to return `super.visitFieldAccess(rexFieldAccess);` in an if-else branch.